brave/brave-ios
# Fix #8073: Add NFT grouping by none/accounts/networks

## Summary of Changes
Same as Portfolio, NFTs can be grouped by `None`/`Accounts`/Networks 
`None`: no grouping no expandable feature 
`Accounts`: grouping by wallet accounts and each group can be collapsed or expanded  
`Networks`: grouping by networks and each group can be collapsed or expanded 


This pull request fixes #8073

## Submitter Checklist:


## Test Plan:

1. Go to brave wallet and unlock your wallet 
2. Tap NFT in portfolio
3. Keep the dropdown as `Collected`
4. Open filter button in the header
5. Observe there is a new option called `Group By` with default value `None`
6. Adjust other parts inside this settings and save the changes. The filtering should work exactly as before. 
7. Click on this toggle to observer there are three options: None/Accounts/Networks
8. Choose `Accounts`
9. Observe that `Hide Unowned` will be changed to `On` (if it was not initially) and disabled 
10. Click `Save Changes`
11. Observe NFTs are now being displayed in groups by `Accounts` and there should be no unowned NFT in any groups. Any group/account that does not have any NFT will be hidden. Check all NFTs should be visible assets since its under `Collected`
12. Open filter then choose `Networks` for `Group By`
13. Observe `Hide Unowned` is now enabled 
14. Click `Save Changed`
15. Observe NFTs are now being displayed in groups by `Network` and there should be no unowned NFT in any groups. Any group/network that does not have any NFT will be hidden.
16. Let's change the dropdown in the header from `Collected` to `Hidden`
17. Repeat step 7-10 and step 12 - 14
18. Observe each `Group By` is applied correctly. and only hidden or spam NFTs are being grouped and displayed. 
19. Make sure hide or unhide NFTs still work in groups


## Screenshots:

![simulator_screenshot_115AFC85-C1EA-4573-9E06-73D170A2326B](https://github.com/brave/brave-ios/assets/1187676/87cf0187-7b93-4c7e-b6d0-1a6e47b9ff7b) | ![simulator_screenshot_5F1FFB16-2D12-4CE9-8E64-71AB89C837B1](https://github.com/brave/brave-ios/assets/1187676/ea5ebfb5-43ee-4f9c-a732-751580303347) | ![simulator_screenshot_93131FF7-451F-4C19-BEE5-8471B38300E9](https://github.com/brave/brave-ios/assets/1187676/92a311aa-20c0-4bdc-9f23-8ecfc651d5a5)
--- | --- | ---
![simulator_screenshot_61F43603-C53D-474F-9662-FCA53558F6F6](https://github.com/brave/brave-ios/assets/1187676/635c0fb5-5e26-4036-b9f2-64ad578774a2) | ![simulator_screenshot_22D48BED-6299-428F-A2A8-56858E4407E8](https://github.com/brave/brave-ios/assets/1187676/fdcb77f2-0016-4ba8-b801-c66a149ba4a6) | ![simulator_screenshot_386D4FDE-C3B3-44B4-B904-CEDF4F0FF3BB](https://github.com/brave/brave-ios/assets/1187676/9f5ba62a-a264-4485-ad75-07a026d16ea3)

## Reviewer Checklist:

  - `QA/(Yes|No)`
  - `bug` / `enhancement`
